cmd/tmsUtils.go

```diff
@@ -0,0 +1,401 @@
+		log.Entry().Infof("- MTA version: %v", mtaVersion)
+		if namedUser != "" {
+			log.Entry().Infof("- named user: %v", namedUser)
+		}
+		log.Entry().Infof("- node name: %v", nodeName)
+	}
+
+	if len(nodeNameExtDescriptorMapping) > 0 {
+		nodes, errGetNodes := communicationInstance.GetNodes()
+		if errGetNodes != nil {
+			log.SetErrorCategory(log.ErrorService)
+			return fmt.Errorf("failed to get nodes: %w", errGetNodes)
+		}
+
+		mtaYamlMap, errGetMtaYamlAsMap := getYamlAsMap(utils, "mta.yaml")
+		if errGetMtaYamlAsMap != nil {
+			log.SetErrorCategory(log.ErrorConfiguration)
+			return fmt.Errorf("failed to get mta.yaml as map: %w", errGetMtaYamlAsMap)
+		}
+		_, isIdParameterInMap := mtaYamlMap["ID"]
+		_, isVersionParameterInMap := mtaYamlMap["version"]
+		if !isIdParameterInMap || !isVersionParameterInMap {
+			var errorMessage string
+			if !isIdParameterInMap {
+				errorMessage += "parameter 'ID' is not found in mta.yaml\n"
+			}
+			if !isVersionParameterInMap {
+				errorMessage += "parameter 'version' is not found in mta.yaml\n"
+			}
+			log.SetErrorCategory(log.ErrorConfiguration)
+			return errors.New(errorMessage)
+		}
+
+		// validate the whole mapping and then throw errors together, so that user can get them after a single pipeline run
+		nodeIdExtDescriptorMapping, errGetNodeIdExtDescriptorMapping := formNodeIdExtDescriptorMappingWithValidation(utils, nodeNameExtDescriptorMapping, nodes, mtaYamlMap, mtaVersion)
+		if errGetNodeIdExtDescriptorMapping != nil {
+			log.SetErrorCategory(log.ErrorConfiguration)
+			return errGetNodeIdExtDescriptorMapping
+		}
+
+		for nodeId, mtaExtDescriptorPath := range nodeIdExtDescriptorMapping {
+			obtainedMtaExtDescriptor, errGetMtaExtDescriptor := communicationInstance.GetMtaExtDescriptor(nodeId, fmt.Sprintf("%v", mtaYamlMap["ID"]), mtaVersion)
+			if errGetMtaExtDescriptor != nil {
+				log.SetErrorCategory(log.ErrorService)
+				return fmt.Errorf("failed to get MTA extension descriptor: %w", errGetMtaExtDescriptor)
+			}
+
+			if obtainedMtaExtDescriptor != (tms.MtaExtDescriptor{}) {
+				_, errUpdateMtaExtDescriptor := communicationInstance.UpdateMtaExtDescriptor(nodeId, obtainedMtaExtDescriptor.Id, mtaExtDescriptorPath, mtaVersion, description, namedUser)
+				if errUpdateMtaExtDescriptor != nil {
+					log.SetErrorCategory(log.ErrorService)
+					return fmt.Errorf("failed to update MTA extension descriptor: %w", errUpdateMtaExtDescriptor)
+				}
+			} else {
+				_, errUploadMtaExtDescriptor := communicationInstance.UploadMtaExtDescriptorToNode(nodeId, mtaExtDescriptorPath, mtaVersion, description, namedUser)
+				if errUploadMtaExtDescriptor != nil {
+					log.SetErrorCategory(log.ErrorService)
+					return fmt.Errorf("failed to upload MTA extension descriptor to node: %w", errUploadMtaExtDescriptor)
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func fileUpload(config tmsUploadOptions, communicationInstance tms.CommunicationInterface, utils tmsUtils) (string, error) {
```

Comment:
[minor]
```suggestion
func uploadFile(config tmsUploadOptions, communicationInstance tms.CommunicationInterface, utils tmsUtils) (string, error) {
```

Reply:
changed

Reply:
I cannot see the change.

Reply:
sorry, renamed now to  tmsUploadFile, uploadFile is used in the package